### DISCUSSION TITLE
How to cache database calls for dynamic routes?

### DISCUSSION THREAD

**Discussion by Turbo-Thorschten** [nuxt/nuxt#15966] [Questions]
Hey!

I have a nuxt app that gets its data from a firebase backend on my api. I have a page page/blog/[blogPostName.vue] which calls the the api. However I don't want to call firebase everytime a user visits a blog post. I want the data loaded on the first visit preferentially and then stay cached. My understanding was that I could use swr for that with route rules:

'/blog/**': { swr: 3600 }

However the server/api still gets called on each visit. Then I tried using defineCachedEventHandler on the api, but this returns the same content of the first visited blogpost to every other post thats visited. So I guess you can only use this for data thats not dependent on the event argument. Another idea I had was to preload all blogposts once during startup and define an interval to update the data. However I have no idea how to do that.

What would be your way to go on this problem?

**Comment by oneart-dev** [comment]
Same questions. No need to make calls to api for every user. It's all the same data for everyone. I just need nuxt to render it once every 10 minutes with all api calls, save it and send to all people. Right now I can't imagine how nuxt can be used for high load projects with cache removing recently from fetch and not possible to cache entire page.

**Comment by Baroshem** [comment]
For users calls, you would probably not use any caching at all. You do not wont to serve one user content of another user that was cached ;)

Have you tried using for example `useAsyncData` and passing a `key` to it? It should allow you to fetch the api call result in a variable so that when you will access the same site, you will get the result from cache?

**Reply by Baroshem** [reply]
https://nuxt.com/docs/api/composables/use-async-data/#params
